### PR TITLE
gamnit: add documentation and standardize the API

### PR DESCRIPTION
**Pull request by xymus**
Improve the documentation for the central services of gamnit, mainly for entities visible from gamnit.org. Also, to improve the API, remove the `Gamnit` from some class name and move a few named constuctors to factories or their own class.

### CODE REVIEW

File: lib/gamnit/depth/more_meshes.nit [hunk 1]
```diff
@@ -80,6 +80,8 @@ class Plane
 end
 
 # Cube, with 6 faces
+#
+# Occupies `[0.5..0.5]` on all three axes.
```
Comment (BlackMinou) [inline lib/gamnit/depth/more_meshes.nit]: [0.5..0.5] ?

File: lib/gamnit/textures.nit [hunk 2]
```diff
@@ -12,19 +12,56 @@
...
+#
+#         # Use the texture
+#         var sprite = new Sprite(texture, new Point3d[Float](0.0, 0.0, 0.0))
+#         app.sprites.add sprite
+#     end
+# end
+# ~~~
+#
+# Otherwise, they can be loaded and error checked explicitly.
+# A texture can be assigned to a sprite before being loaded.
```
Comment (BlackMinou) [inline lib/gamnit/textures.nit]: `can` or `can't` ? Since you don't assign it to a sprite in the example right below, I'm confused

**Comment by xymus** [comment]
Fix for @BlackMinou's comments.

File: lib/gamnit/flat.nit [hunk 3]
```diff
@@ -45,11 +45,53 @@ import gamnit::dynamic_resolution
...
+# # Place it a bit off the top left of the screen
+# var pos = app.ui_camera.top_left.offset(128.0, -128.0, 0)
+#
+# # Load texture and create sprite
+# texture = new Texture("path/in/assets.png")
+# sprite = new Sprite(texture, pos)
+#
+# # Add it to the UI (above world sprites)
+# app.ui_sprites.add sprite
+# ~~~
```
Comment (privat) [inline lib/gamnit/flat.nit]: is there a bug in the docunit?

http://gresil.org/jenkins/view/Int%C3%A9gration/job/nit_test_tools/1170/testReport/nitunit.gamnit/Sprite/_class_/

**Comment by xymus** [comment]
Rebased to fix the unit tests.

**Pull request merged**